addons/cert.py

```diff
@@ -234,6 +245,101 @@ def msc30(data):
         if simpleMatch(token, "rand ( )") and isStandardFunction(token):
             reportError(token, 'style', 'Do not use the rand() function for generating pseudorandom numbers', 'MSC30-c')
 
+# STR03-C
+# Do not inadvertently truncate a string
+def str03(data):
+    for token in data.tokenlist:
+        if token.str=='strncpy':
+            nextToken = token.astParent
+            while(nextToken):
+                if nextToken.str=='sizeof':
```

Comment:
you should probably check if the 3rd argument is a sizeof call.. not just if sizeof is used anywhere after the strncpy token in the whole translation unit.